apps/browser/settings/searchenginemodel.h

```diff
@@ -21,13 +21,21 @@ class SearchEngineModel : public QAbstractListModel, public QQmlParserStatus
     Q_OBJECT
     Q_INTERFACES(QQmlParserStatus)
 
+    Q_ENUMS(Status)
+
     Q_PROPERTY(int count READ rowCount NOTIFY countChanged)
 
 public:
     enum Roles {
         UrlRole = Qt::UserRole + 1,
         TitleRole,
-        InstalledRole
+        StatusRole
+    };
+
+    enum Status {
+        System,
+        Available,
+        Installed
```

Comment:
Maybe "UserInstalled" as then this would be more clearly separated from the system one.